src/senaite/sync/browser/views.py

```diff
@@ -427,6 +472,16 @@ def get_authenticated_user(self):
         """
         return self.get_first_item("users/current")
 
+    def get_registry_records_by_keyword(self, keyword=None):
+        """Return the values of the registry records
+        associated to the specified keyword in the source instance.
+        If keyword is None it returns the whole registry
+        """
+        if keyword is None:
+            return self.get_json("registry")
+
+        return self.get_json("registry/{}".format(keyword))
```

Comment:
Refactor this function to return immediately the registry `items`. Maybe you can use the `get_items` method or even the `yield_items` to get the items of all pages.

Reply:
@ramonski The problem we will face by doing this is that then we won't have access to the api url for those items

Reply:
@ramonski I have refactored the method to directly return `items` and for now I have removed the API url column for registry records